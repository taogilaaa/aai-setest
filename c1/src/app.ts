import express from 'express';
import multer from 'multer';

const app = express();
const upload = multer();

app.post('/query', upload.any(), (request, response) => {
  if (!Array.isArray(request.files) || request.files.length === 0) {
    return response.status(400).send({
      error: {
        code: 'BAD_REQUEST',
        message: 'Missing or invalid parameters',
      },
    });
  }

  const [firstImage] = request.files;
  const imageSize = firstImage.size / Math.pow(1024, 1);

  return response.status(200).send({
    status: 'SUCCESS',
    size: `${imageSize.toFixed(2)}kb`,
  });
});

export { app };
