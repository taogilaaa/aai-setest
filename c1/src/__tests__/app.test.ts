import request from 'supertest';
import { app } from '../app';
import path from 'path';

describe('query controller', () => {
  it('should return success and size on proper request', async () => {
    const filePath = path.join(__dirname, '../../assets/github.png');

    const response = await request(app)
      .post('/query')
      .attach('image', filePath);

    expect(response.body).toMatchSnapshot();
  });

  it('should return BAD_REQUEST on missing image', async () => {
    const response = await request(app).post('/query');

    expect(response.body).toMatchSnapshot();
  });
});
