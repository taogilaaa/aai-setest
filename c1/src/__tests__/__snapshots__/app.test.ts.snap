// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query controller should return BAD_REQUEST on missing image 1`] = `
Object {
  "error": Object {
    "code": "BAD_REQUEST",
    "message": "Missing or invalid parameters",
  },
}
`;

exports[`query controller should return success and size on proper request 1`] = `
Object {
  "size": "6.94kb",
  "status": "SUCCESS",
}
`;
